shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -646,6 +682,16 @@ - (void)setMarkedText:(id)string
   if (!_activeModel->composing()) {
     _activeModel->BeginComposing();
   }
+
+  if (replacementRange.location != NSNotFound) {
+    _activeModel->SetComposingRange(
```

Comment:
This doesn't seem to be correct according to the documentation https://developer.apple.com/documentation/appkit/nstextinputclient/1438246-setmarkedtext?language=objc

> replacementRange
The range to replace, computed from the beginning of the marked text.

Reply:
Can you be more specific? I set the replacement range as composing range, that way it will be replaced in `TextInputModel::UpdateComposingText`.

Reply:
It says `replacementRange` is computed from the beginning of the marked text so it sounds like the starting point should be calculated from the current marked text.

Reply:
`replacementRange` is specified as being computed relative to the beginning of the marked text (as opposed to from the beginning of the entire text).

My memory is a bit fuzzy since it's been a while since I hacked on this, but I believe the range to replace is:

start: markedRange.location + replacementRange.location
end: markedRange.location + replacementRange.location + replacementRange.length

Reply:
I don't see how that would work. With character alternates selection 
<img width="262" alt="Screenshot 2022-06-23 at 06 16 04" src="https://user-images.githubusercontent.com/96958/175207818-5acbe205-a364-4f31-898d-60aceef9cd57.png">
`- (void)setMarkedText:selectedRange:replacementRange:` is first called without composing being active and with `replacementRange` provided. So at that point _there is no marked range_. This happens when you select first candidate. The offset specified is clearly from the beginning of the string (it replaces the long pressed character).

I looked at what [Chromium does](https://source.chromium.org/chromium/chromium/src/+/main:content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm;drc=e4e66f3618a6a2a2d05d87ab98431045baaf3006;l=1976) and as far as I can tell it also sets the marked range to replacement range (from the beginning of string) - seems same as what this PR does. 

And this is what [webkit does](https://github.com/WebKit/WebKit/blob/bfe3f272a5b8cc261b68249303d357fd424ec572/Source/WebKit/WebProcess/WebPage/WebPage.cpp#L6135). The `replacementEditingRange` argument is the `replacementRange`, and the whole thing is set as selection before updating as text.

I think the documentation might be wrong about this one, at very least incomplete since the `replacementRange` is provided in situation when there is no marked range.

Reply:
Agreed the documentation is so ambiguous as to be close to useless. When I wrote the initial UIKit equivalent, the Apple APIs there were at least a little cleaner, but docs weren't much better and much of it was figured out purely by manually checking the behaviour :/

Curious -- what's the behaviour is in a case where this is called where there is existing mark text? I'm assuming that would get called when, for example hiragana is converted to a kanji candidate with Japanese input? To test:
1. Type "hello" so you're offset from the zero position of the string
2. switch to Japanese input
3. set hiragana mode from the input dropdown
4. type nihonngo (you should get にほんご)
5. press space bar and select 日本語 (it may convert automatically without the space depending on your input history)

I'm guessing that during the hiragana input, this might get called multiple times as the conversions happen. Are the replacementRange values always relative to 0 in those cases, or just when there's no mark region? Agreed it would be weird to have two behaviours depending on the presence of an existing mark region, but curious. 

Assuming that behaves well with a CJK mark region, then seems reasonable. Thanks for checking.

Reply:
The `replacementRange` with any kind of CJK composition always seems to be `NSNotFound`. I tried your example and it was `NSNotFound` as well. The only way I can trigger a non-empty `replacementRange` is the long press character alternate selection, which starts with no composition and then switches to composition upon first selection. Even here the `replacementRange` is only used initially (when not compositing), and in subsequent selection changes the `replacementRange` is empty and (now non-empty)`markedRange` is used.

Reply:
Done.